### PR TITLE
WIP Build an API endpoint to list products in the shopfront (move code from shop/products to api/products/shop)

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Relates to #4042 

#### What should we test?
TODO

#### Release notes
Changelog Category: Added
Added new API endpoint to list products in a given shop

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
I added a quick comment in #4271 that's relevant to this API endpoint. It should help a lot here.

What do you think to moving in this direction (filter and paginate before scoping variants):
https://github.com/openfoodfoundation/openfoodnetwork/compare/master...Matt-Yorkley:shop_endpoint

File: app/controllers/api/products_controller.rb [hunk 1]
```diff
@@ -8,6 +8,42 @@ class ProductsController < Api::BaseController
 
     skip_authorization_check only: [:show, :bulk_products, :overridable]
 
+    def shop
+      renderer = ProductsRenderer.new(current_distributor, current_order_cycle)
+      products = ProductsFilterer.new(renderer.products).call
```
Comment (Matt-Yorkley) [inline app/controllers/api/products_controller.rb]: The method on the renderer is `#products_json`

Comment (luisramos0) [reply app/controllers/api/products_controller.rb]: it was json... now that we dont have cache I dont think we should waste time parsgin and unparsing json... some work to do here.

**Comment by luisramos0** [comment]
closing, this was done in #4334

**Pull request closed without merging**